evm/vm/state_transition_helper.py

```diff
@@ -0,0 +1,47 @@
+import copy
+import logging
+import traceback
+
+
+logger = logging.getLogger("evm.research.state_transition_helper")
+
+
+def apply_transaction(transaction, block, prev_vm, chaindb):
```

Comment:
Are we sure that this is actually a pure function? It's clear that prev_vm is not being modified; but could chaindb be modified anywhere? Likely not but I'm not well-versed enough in the code to be able to tell.

Reply:
Currently, I only checked `state_root` hasn't been changed (`chain.get_vm()` uses `chain.chaindb` to initialize `VM`):
https://github.com/ethereum/py-evm/pull/235/files#diff-555a9b3a5d79c67dccb0a54b0f1649ddR51
I will add more tests after #236 .